ros2cli/ros2cli/node/daemon.py

```diff
@@ -65,13 +67,16 @@ def __exit__(self, exc_type, exc_value, traceback):
         self._proxy.__exit__(exc_type, exc_value, traceback)
 
 
-def is_daemon_running(args):
+def is_daemon_running(args, timeout=None):
     """
     Check if the daemon node is running.
 
     :param args: `DaemonNode` arguments namespace.
+    :param timeout: option duration, in seconds, to wait
+      for the daemon node to respond. If it is not given,
+      the global default timeout setting is used.
```

Comment:
I can't think of any reasonable circumstances where it would take longer than that either. I'm in favor of a default timeout as well, but I think we need more feedback on what value might be appropriate.

I also think we need to improve the error message if it becomes the default behavior. Right now, the exception is unhandled.